src/components/TextField/TextField.tsx

```diff
@@ -11,19 +11,49 @@ import { IProps as InputProps } from '../Input'
 import { ITextAreaProps } from '../TextArea'
 
 // Styles
-import { TextFieldBase, BASE_CLASS_NAME } from './TextField.styled'
+import { TextFieldBase, TextFieldHelpersWrapper, BASE_CLASS_NAME } from './TextField.styled'
 
 interface ITextFieldBaseProps {
   label?: string
   helperText?: string
   error?: boolean
   fullWidth?: boolean
   textArea?: boolean
+  minLength?: number
+  maxLength?: number
 }
 
 interface TextFieldInputProps extends ITextFieldBaseProps, InputProps {}
 interface TextFieldAreaProps extends ITextFieldBaseProps, ITextAreaProps {}
 
+type ILength = {
+  min?: number | undefined
+  max?: number | undefined
+}
+
+interface IValidateLength {
+  length: ILength
+  value: string | number | undefined
+}
+
+const ValidateLength = ({ length, value }: IValidateLength): string | null => {
+  const { min, max } = length
+
+  if (min) {
+    if (value !== '' && String(value).length < min) {
+      return `Min length ${min} characters`
```

Comment:
I think instead of showing a message, we should only show a counter like:

0/100
100/100

Then if is 100/100 or max/max you don't allow them to add more, also you can have black color for the numbers until reach 50%, then after that every 10% more you can change the color to yellow, orange and when is 90/100 you put red.

Check how is the text counter I did in the old UI project https://github.com/ContentPI/ui/blob/main/src/TextCounter/index.tsx